js/query_syntax/filter.js

```diff
@@ -0,0 +1,66 @@
+function GenerateAggregateOrderBy(options = {}) {
+	return [
+		Keyword("ORDER"),
+		Keyword("BY"),
+		GenerateOrderTerms()
+	]
+}
+
+function GenerateFilterClause(options = {}) {
+	return [
+		Keyword("FILTER"),
+		Keyword("("),
+		Keyword("WHERE"),
+		Expression('expr'),
+		Keyword(")")
+	]
+}
+
+function GenerateAggregate(options = {}) {
+	return Diagram([
+		AutomaticStack([
+			Expression("aggregate-name"),
+			Keyword("("),
+			Choice(0, [
+				Sequence([
+					Optional(Keyword("DISTINCT"), "skip"),
+					OneOrMore(Sequence([
+						Expression()
+					]), Keyword(",")),
+					Optional(Expandable("order-by", options, "order-by", GenerateAggregateOrderBy))
+				]),
+				new Skip()
+			]),
+			Keyword(")"),
+			Optional(Expandable("filter-clause",options,"filter-clause",GenerateFilterClause))

```

Comment:
Does this need to be expandable? There is plenty of room.

Reply:
Fixed!